code/datums/elements/openspace_item_click_handler.dm

```diff
@@ -19,7 +19,12 @@
 	SIGNAL_HANDLER
 	if(target.z == user.z)
 		return
-	var/turf/turf_above = get_step_multiz(target, UP)
-	if(turf_above?.z == user.z)
-		INVOKE_ASYNC(source, TYPE_PROC_REF(/obj/item, handle_openspace_click), turf_above, user, user.CanReach(turf_above, source), click_parameters)
+	var/turf/checked_turf = get_turf(target)
+	do
+		checked_turf = checked_turf?.above()
+		if(checked_turf?.z == user.z)
+			INVOKE_ASYNC(source, TYPE_PROC_REF(/obj/item, handle_openspace_click), checked_turf, user, user.CanReach(checked_turf, source), click_parameters)
+			break
+	while(!isnull(checked_turf))
```

Comment:
Is there a difference between doing it this way and:
```
	while(!isnull(checked_turf))
		checked_turf = checked_turf?.above()
		if(checked_turf?.z == user.z)
			INVOKE_ASYNC(source, TYPE_PROC_REF(/obj/item, handle_openspace_click), checked_turf, user, user.CanReach(checked_turf, source), click_parameters)
			break
```

It's jus' a metter of consistency with other while loops really.

Reply:
Should work the same, I just wanted to avoid the check on the first iteration where you _know_ it isn't already null. Updated for consistency